src/components/views/messages/TextualBody.js

```diff
@@ -37,28 +38,76 @@ module.exports = React.createClass({
 
         /* link URL for the highlights */
         highlightLink: React.PropTypes.string,
+
+        /* callback for when our widget has loaded */
+        onWidgetLoad: React.PropTypes.func,
+    },
+
+    getInitialState: function() {
+        return {
+            link: null,
```

Comment:
I think I'd missed that calculating `link` depended on the DOM having been created, and you're absolutely right.